lib/src/sdk/dart.dart

```diff
@@ -50,5 +73,17 @@ class DartSdk extends Sdk {
   String get rootDirectory => _rootDirectory;
 
   @override
-  String? packagePath(String name) => null;
+  String? packagePath(String name) {
+    if (!isAvailable) return null;
```

Comment:
We don't need it _right now_ but it still felt right to leave it in, but I am also happy to remove it if you prefer. It seems highly unlikely this would ever be false.

Reply:
Maybe we should factor it out, such that Sdk.packagePath can assume that available is true (for all sdks)

Reply:
Sure.

https://github.com/dart-lang/pub/blob/9c041f97c07ea17f1460f2d7df5aa86b63a72917/lib/src/source/sdk.dart#L118 makes the call to `.packagePath()`. It could preface it with a call to `isAvailable()`, instead of baking that logic into each `.packagePath()` implementation.

If it is too much, we can also do the refactor later. Feel free to land this if you are happy.

Reply:
I think a follow up would be good if you want to make that change, I am not sure pushing that to the consumer is best though?